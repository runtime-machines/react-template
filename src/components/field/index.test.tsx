import * as React from 'react';
import { shallow } from 'enzyme';

import { Field } from '.';

describe('Field component', () => {
	it('should render successfully', () => {
		const tree = shallow(
			<Field
				register={(): jest.Mock => jest.fn()}
				type="email"
				name="email"
				label="Email Address"
				error={{}}
				placeholder="someone@example.com"
			/>
		);

		expect(tree).toMatchSnapshot();
	});
});
