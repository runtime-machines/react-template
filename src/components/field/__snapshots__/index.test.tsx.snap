// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field component should render successfully 1`] = `
<div
  className="c-form__field c-form__field--email c-form__field--error"
>
  <header
    className="c-form__field-head"
  >
    <label
      htmlFor="email"
    >
      Email Address
    </label>
    <p
      className="c-form__error"
    />
  </header>
  <input
    id="email"
    name="email"
    placeholder="someone@example.com"
    type="email"
  />
</div>
`;
